aea/multiplexer.py

```diff
@@ -263,41 +263,58 @@ async def disconnect(self) -> None:
         async with self._lock:
             if self.connection_status.is_disconnected:
                 self.logger.debug("Multiplexer already disconnected.")
-                await asyncio.wait_for(self._stop(), timeout=60)
```

Comment:
@solarw why was this needed in the first place?
Just making the observation in case this was here for a reason (to handle an edge case for e.g.)

I understand that in the previous version disconnecting all connections and then cancelling background coroutines  *in this order* was problematic. However they were still both executed after a call of `disconnect()`.

Reply:
@lrahmani i think there is no reason to call stop second time if already disconnected, this produced me an error in tests after the refactoring. now all the tests are passing ok, so not sure it's required anymore.